crates/build/src/lib.rs

```diff
@@ -215,6 +232,56 @@ fn construct_workdir(app_dir: &Path, workdir: Option<impl AsRef<Path>>) -> Resul
     Ok(cwd)
 }
 
+/// Saves the build profile to the "last build profile" file.
+/// Errors are ignored as they should not block building.
+pub fn save_last_build_profile(app_dir: &Path, profile: Option<&str>) {
+    let app_stash_dir = app_dir.join(".spin");
+    _ = std::fs::create_dir_all(&app_stash_dir);
+    let last_build_profile_file = app_stash_dir.join(LAST_BUILD_PROFILE_FILE);
+    _ = std::fs::write(
+        &last_build_profile_file,
+        profile.unwrap_or(LAST_BUILD_ANON_VALUE),
+    );
+}
+
+/// Reads the last build profile from the "last build profile" file.
+/// Errors are ignored.
+pub fn read_last_build_profile(app_dir: &Path) -> Option<String> {
+    let app_stash_dir = app_dir.join(".spin");
+    let last_build_profile_file = app_stash_dir.join(LAST_BUILD_PROFILE_FILE);
+    let last_build_str = std::fs::read_to_string(&last_build_profile_file).ok()?;
+
+    if last_build_str == LAST_BUILD_ANON_VALUE {
+        None
+    } else {
+        Some(last_build_str)
+    }
```

Comment:
Not _inarguably_ better, but another option:


```suggestion
    (last_build_str != LAST_BUILD_ANON_VALUE).then_some(last_build_str)
```

Reply:
Maybe it's too much Go programming but I find this a bit obscure and would prefer to keep spelling it out.  Appreciate the education though - just give me a couple of years to internalise it!